File: TP2/Parte3/src/shard_config.js
use admin

db.runCommand({addshard:"localhost:43234", name:"shard43234"});
db.runCommand({addshard:"localhost:43235", name:"shard43235"});
db.runCommand({addshard:"localhost:43236", name:"shard43236"});
db.runCommand({addshard:"localhost:43237", name:"shard43237"});
db.runCommand({addshard:"localhost:43238", name:"shard43238"});
db.runCommand({addshard:"localhost:43239", name:"shard43239"});
db.runCommand({addshard:"localhost:43240", name:"shard43240"});

use test_sharding
sh.enableSharding("test_sharding")

db.people.ensureIndex({"_id": "hashed"})
sh.shardCollection("test_sharding.people", {"_id": "hashed"})

load("crear_inserts.js")

use admin
db.runCommand({removeShard: "shard43234"})
db.runCommand({removeShard: "shard43235"})
db.runCommand({removeShard: "shard43236"})
db.runCommand({removeShard: "shard43237"})
db.runCommand({removeShard: "shard43238"})
db.runCommand({removeShard: "shard43239"})
db.runCommand({removeShard: "shard43240"})

use test_sharding
db.dropDatabase()

File: TP2/Parte3/src/shard_config.cpp
#include <iostream>
#include <cstdio>
#include <stdlib.h>
#include <assert.h>
using namespace std;

int main ( int argc, char *argv[] ) {
	
	int shards = atoi(argv[1]);
	int puerto_inicial = atoi(argv[2]);
	string tipo = argv[3];
	
	assert(tipo == "simple" || tipo == "hashed");
	if (tipo == "simple")
		tipo = "{\"zipcode\": 1}";
	else {
		tipo = "{\"_id\": \"hashed\"}";
	}

	freopen ("shard_config.js", "w", stdout);

	cout << "use admin" << endl; cout << endl;
	
	for (int i = puerto_inicial; i < puerto_inicial + shards; i++) {
		cout << "db.runCommand({addshard:\"localhost:" << i << "\", name:\"shard" << i << "\"});" << endl;
	}
	cout << endl;
	cout << "use test_sharding" << endl;
	cout << "sh.enableSharding(\"test_sharding\")" << endl; cout << endl;

	cout << "db.people.ensureIndex("<< tipo <<")" << endl; // aca hay que poner opcion hashed
	cout << "sh.shardCollection(\"test_sharding.people\", "<< tipo <<")" << endl; cout << endl;
	cout << "load(\"crear_inserts.js\")" << endl; cout << endl;

	cout << "use admin" << endl;
	for (int i = puerto_inicial; i < puerto_inicial + shards; i++) {
		cout << "db.runCommand({removeShard: \"shard" << i << "\"})" << endl;
	}
	cout << endl;

	cout << "use test_sharding" << endl;
	cout << "db.dropDatabase()" << endl;
	return 0;
}
